src/Microsoft.Azure.WebJobs.Host/Singleton/SingletonListener.cs

```diff
@@ -101,6 +98,10 @@ public void Dispose()
                 LockTimer.Dispose();
             }
 
+            // When we Dispose, it's important that we release the lock if we
+            // have it.
+            ReleaseLockAsync().Wait();
```

Comment:
Could this result in deadlocks? Sync context, etc. 

We may need to do this:
  var theTask = Task.Run( ()=> ReleaseLockAsync());
  theTask.GetAwaiter().GetResult();